### PR TITLE
add purge cache

### PR DESCRIPTION
**Pull request by DavidCramer**

### CODE REVIEW

File: php/assets/class-rest-assets.php [hunk 1]
```diff
@@ -105,30 +95,72 @@ public function rest_endpoints( $endpoints ) {
...
+	public function rest_purge_all( $request ) {
+
+		$count         = $request->get_param( 'count' );
+		$asset_parent  = (int) $request->get_param( 'parent' );
+		$transient_key = '_purge_cache' . $asset_parent;
+		$query_args    = array(
+			'post_type'              => Assets::POST_TYPE_SLUG,
+			'posts_per_page'         => 1,
+			'paged'                  => 1,
+			'post_status'            => array( 'inherit', 'draft', 'publish' ),
```
Comment (pereirinha) [inline php/assets/class-rest-assets.php:136]: @DavidCramer 

I'm a bit confused here. What's does the `publish` state stands for?
I thought that we only had 2 states as per this [comment](https://github.com/cloudinary/cloudinary_wordpress/pull/484/files#diff-12a132c16918f2f8786165885c22904fba05f2c21467113a4fda1dbeed882458R164-R165).

**Review by pereirinha — COMMENTED** [review]
@DavidCramer 

This is looking good. And I do have a couple of questions for you.

Thanks.

File: php/class-assets.php [hunk 2]
```diff
@@ -363,14 +400,33 @@ public function meta_updates() {
 	 * @hook cloudinary_string_replace
 	 */
 	public function add_url_replacements() {
+		$clear = filter_input( INPUT_GET, 'cloudinary-cache-clear', FILTER_SANITIZE_STRING );
+		if ( $clear ) {
```
[the remark this answers is not in the document]
Comment (DavidCramer) [reply php/class-assets.php:404]: We're checking if we get anything. If it's not set it's null.

File: php/class-assets.php [hunk 3]
```diff
@@ -363,14 +400,33 @@ public function meta_updates() {
...
+			$referrer = filter_input( INPUT_SERVER, 'HTTP_REFERER', FILTER_SANITIZE_URL );
+			if ( $this->asset_ids ) {
+				foreach ( $this->asset_ids as $asset_id ) {
+					wp_delete_post( $asset_id );
```
[the remark this answers is not in the document]
Comment (DavidCramer) [reply php/class-assets.php:408]: Very good point!

File: php/assets/class-rest-assets.php (hunk 1 above)
Comment (DavidCramer) [reply php/assets/class-rest-assets.php:136]: Publish is ignored withing the media post types, so publish is given to the parent post.

File: php/class-assets.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (DavidCramer) [reply php/class-assets.php]: @pereirinha - I've. changed it to be a nonce instead, which addresses the next comment.

File: php/class-assets.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (DavidCramer) [reply php/class-assets.php:407]: @pereirinha - as mentioned, addressed in the previous comment.

**Pull request merged**